### ISSUE TITLE
how to save one struct has pointer  and embedded map with cbor 

### ISSUE THREAD

**Issue by mullerhai** [fxamacker/cbor#80]
I have two  struct  like this 

type IdsMapping struct {
	UserIdIndexDict map[float64]float64
	IndexUserIdDict map[float64]float64
	ItemIdIndexDict map[float64]float64
	indexItemIdDict map[float64]float64
}

type LFM struct {
	classCount             int
	iterCount              int
	featureName            string
	labelName              string
	lr                     float64
	lam                    float64
	UserItemRatingMatrix   *mat.Dense
	ModelPfactor           *mat.Dense
	ModelQfactor           *mat.Dense
	RatingDf               *dataframe.DataFrame
	UseridItemidDict       map[float64]map[float64]float64
	UseridSet              *gset.Set
	ItemidSet              *gset.Set
	IdsMapping             *IdsMapping
	UserIndexItemIndexDict map[float64]map[float64]float64
}

I want to use cbor  to serilazition LFM struct ,how to use it ,because the struct is complex ,thx

**Comment by fxamacker** [comment]
It should work.

Pointers -- this library will correctly handle nil and not nil pointers for both encoding and decoding. For example, it will encode the data pointed to by the pointer.

Nested maps -- this library will correctly handle nested maps up to 32 levels of nesting. For example, map[float64]map[float64]float64 is only 2 levels of nesting so it's OK.

Unless gset.Set, mat.Dense, dataframe.DataFrame contain data that exceeds 32 levels of nesting, then they should be OK.

Unexported fields -- like encoding/json, unexported fields (lower case name) are not serialized. For example, featureName won't be serialized unless you rename it to FeatureName.

Try this as a starting point:

```
var v LFM

// TODO: populate v here

// Encode v to CBOR
dataAsCBOR, err := cbor.Marshal(v, cbor.EncOptions{}); 
if err != nil {
	return err
}
```

See the [Usage section](https://github.com/fxamacker/cbor#usage) and also [example_test.go](https://github.com/fxamacker/cbor/blob/master/example_test.go) for more info.

**Comment by x448** [comment]
It's been 2 weeks, can this be closed?

**Comment by fxamacker** [comment]
@mullerhai Hi, it's been 18 days so I'm closing this.  Please open an issue if you have more questions.  

I noticed you use float64.  I released v1.5 today that which adds support for shrinking 
float64 -> float32 -> float16 if the values can round-trip to make your serialized data size smaller.

https://github.com/fxamacker/cbor/releases
